LiskHQ/lisk-desktop
# Store custom node URL permanently - Closes #3762

### What was the problem?

This PR resolves #3762

### How was it solved?
There are 2 sets of changes:
- The rest of the commits clean up the Login screen component since this component was changed over and over based on the design and infrastructure changes.  I removed 30% of the code in screens/login/login.js and it still works as expected.
- I cleaned up the network selector by creating single-purpose components which are easier to read and maintain.
- Added a new property in network state to store the latest custom network address. Tested the relevant actions and reducers.

### How was it tested?
 - Increased unit test coverage in relevant utility files.
 - Removed Chai where encountered. 
 - Tested the changes manually too:
    - Try connecting to a custom node.
    - Open the network switcher again, the network you connected must be saved.
    - Hover over the network name at the topbar, it should show the network address.
    - Change to testnet and/or mainnet.
    - Open network switcher again, the latest custom network address must still be visible.
    - You should be able to edit and delete the saved custom network address.
    - Make sure App won't throw if the settings wasn't saved previously in the localstorage.
